### PR TITLE
Requested Plugins Bundle

### PR DESCRIPTION
**Pull request by giorod3**
Bundle of plugins requested by enterprise user

### CODE REVIEW

**Review by matthewdfuller — CHANGES_REQUESTED** [review]

File: plugins/aws/ec2/allowedCustomPorts.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Open Custom Ports',
```
Comment (giorod3) [inline plugins/aws/ec2/allowedCustomPorts.js:5]: Allowed Custom Ports

Comment (giorod3) [inline plugins/aws/ec2/allowedCustomPorts.js:5]:
```suggestion
    title: 'Allowed Custom Ports',
```

File: plugins/aws/ec2/allowedCustomPorts.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    category: 'EC2',
+    description: 'Ensures that the defined ports are not exposed publicly',
+    more_info: 'Security groups should be used to restrict access to ports from known networks.',
+    link: 'https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html',
+    recommended_action: 'Modify the security group to ensure the ports are not exposed publicly.',
+    apis: ['EC2:describeSecurityGroups'],
+    settings: {
+        open_port_allowed_list: {
+            name: 'EC2 Allowed Open Ports',
+            description: 'A comma-delimited list of ports that indicates open ports allowed for any connection',
```
Comment (matthewdfuller) [inline plugins/aws/ec2/allowedCustomPorts.js:15]: @giorod3 - Add an example value here in the description

File: plugins/aws/ec2/allowedCustomPorts.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            regex: '[a-zA-Z0-9,:]',
+            default: 'tcp:80,tcp:443'
```
Comment (matthewdfuller) [inline plugins/aws/ec2/allowedCustomPorts.js:17]: @giorod3 - And change this to an empty string default

File: plugins/aws/ec2/allowedCustomPorts.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+        var awsOrGov = helpers.defaultPartition(settings);
+
```
Comment (matthewdfuller) [inline plugins/aws/ec2/allowedCustomPorts.js:26]: And around here, exit the plugin if settings value is empty string

File: plugins/aws/s3/bucketPublicAccessBlock.js [hunk 5]
```diff
@@ -3,37 +3,43 @@ var helpers = require('../../../helpers/aws');
...
+            default: true
         }
     },
 
     run: function(cache, settings, callback) {
         var config = {
-            s3_public_access_block_allow_pattern: settings.s3_public_access_block_allow_pattern || this.settings.s3_public_access_block_allow_pattern.default
+            s3_public_access_block_allow_pattern: settings.s3_public_access_block_allow_pattern || this.settings.s3_public_access_block_allow_pattern.default,
+            check_global_block: (settings.check_global_block === false) ? false : this.settings.check_global_block.default
         };
```
Comment (matthewdfuller) [inline plugins/aws/s3/bucketPublicAccessBlock.js:30]: Add
```
config. check_global_block = (config. check_global_block == 'true');
```

File: plugins/aws/s3/bucketPublicAccessBlock.js [hunk 6]
```diff
@@ -3,37 +3,43 @@ var helpers = require('../../../helpers/aws');
...
             name: 'S3 Public Access Block Allow Pattern',
             description: 'When set, whitelists buckets matching the given pattern. Useful for overriding buckets outside the account control.',
             regex: '^.{1,255}$',
             default: false
+        },
+        check_global_block: {
+            name: 'Check Global Block',
+            description: 'When set, check account level public access for S3 and override bucket level public access check.',
+            regex: '^(true|false)$',
+            default: true
```
Comment (matthewdfuller) [inline plugins/aws/s3/bucketPublicAccessBlock.js:22]: For consistency change to `'true'`

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
@giorod3 -For s3 public access block, still loop through every bucket and create a passing result if global access block is set.

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
@giorod3 - need to add the check for 'true' vs 'false' to boolean conversion at the top of `run` rather than doing `=== 'true'` later.

**Pull request merged**